lib/Transforms/Tapir/TapirToTarget.cpp

```diff
@@ -160,12 +159,16 @@ bool LowerTapirToTarget::runOnModule(Module &M) {
 
   // Add functions that detach to the work list.
   SmallVector<Function *, 4> WorkList;
-  for (Function &F : M)
+  for (Function &F : M) {
+    if (F.getName() == "main")
+      WorkList.push_back(&F);
+
```

Comment:
I feel like a main-specific pass should be a separate function in TapirTarget

Reply:
This change is actually a bug fix to the Tapir lowering pass.  Some Tapir targets already contain code that expects to see `main` among the functions to process.  But if `main` itself did not contain a `detach`, then this routine in TapirToTarget would prevent those Tapir targets from properly process `main`, leading to incorrect behavior.  We can separate out `main`-specific processing, but we'll need to fix some specific targets as well.  I think that change makes sense to put in another PR.